nixos/tests/dokuwiki.nix

```diff
@@ -70,6 +71,18 @@ in {
     machine.wait_for_open_port(80)
 
     machine.succeed("curl -sSfL http://site1.local/ | grep 'DokuWiki'")
+    machine.fail("curl -sSfL 'http://site1.local/doku.php?do=login' | grep 'Login'")
+
     machine.succeed("curl -sSfL http://site2.local/ | grep 'DokuWiki'")
+    machine.succeed("curl -sSfL 'http://site2.local/doku.php?do=login' | grep 'Login'")
+
+    users_file = open("/tmp/users.auth.php", "a")
+    users_file.write(
+        "admin:$2y$10$ijdBQMzSVV20SrKtCna8gue36vnsbVm2wItAXvdm876sshI4uwy6S:Admin:admin@example.test:user\n"
+    )
+    users_file.close()
+    machine.succeed(
+        "curl -sSfL --data 'u=admin' --data 'p=password' 'http://site2.local/doku.php?do=login' | grep 'Admin'"
```

Comment:
This doesn't work, it catches `Admin` from the error: `User authentication is temporarily unavailable. If this situation persists, please inform your Wiki Admin.`

Reply:
Ok, I think I got it working... but I had to specify the usersFile like this:

```
let

  usersFile = pkgs.writeText "users.auth.php" ''
    admin:$2y$10$ijdBQMzSVV20SrKtCna8gue36vnsbVm2wItAXvdm876sshI4uwy6S:Admin:admin@example.test:user
  '';
...
in
{
...
    services.dokuwiki."site2.local" = {
      usersFile = toString usersFile;
...
```
and then check it like this:
```
...
    machine.succeed(
        "curl -d 'u=admin&p=password' --cookie-jar cjar http://site2.local/doku.php?do=login",
        "curl --cookie cjar --cookie-jar cjar http://site2.local/doku.php?do=login | grep 'Logged in as: koko'",
    )
```

Reply:
Using the cookie jar seems like the sensible option.

`/tmp/users.auth.php` did not work because the file was lost somewhere between the `testScript` and `machine.succeed`. Using `machine.succeed` to write the file works. I would prefer this in favor of the approach involving `let usersFile`, since in that case the file would be write-only, which is different from production.

Reply:
Great job!